matchbox_server/src/main.rs

```diff
@@ -32,39 +26,51 @@ async fn main() {
                 .with_target(false),
         )
         .init();
+}
+
+#[tokio::main]
+async fn main() {
+    // Initialize logger
+    setup_logging();
 
     // Parse clap arguments
     let args = Args::parse();
 
     // Setup router
-    let server_state = Arc::new(futures::lock::Mutex::new(ServerState::default()));
-    let app = Router::new()
-        .route("/health", get(health_handler))
-        .route("/", get(ws_handler))
-        .route("/:room_id", get(ws_handler))
-        .layer(
-            // Allow requests from anywhere - Not ideal for production!
-            CorsLayer::new()
-                .allow_origin(Any)
-                .allow_methods(Any)
-                .allow_headers(Any),
-        )
-        .layer(
-            // Middleware for logging from tower-http
-            TraceLayer::new_for_http().on_response(
-                DefaultOnResponse::new()
-                    .level(Level::INFO)
-                    .latency_unit(LatencyUnit::Micros),
-            ),
-        )
-        .with_state(server_state);
-
-    // Run server
     info!("Matchbox Signaling Server: {}", args.host);
-    axum::Server::bind(&args.host)
-        .serve(app.into_make_service_with_connect_info::<SocketAddr>())
+
+    let mut state = ServerState::default();
+    let server = SignalingServerBuilder::new(args.host, MatchmakingDemoTopology, state.clone())
+        .on_connection_request({
+            let mut state = state.clone();
+            move |connection| {
+                let room_id = RoomId(connection.path.clone().unwrap_or_default());
+                let next = connection
+                    .query_params
+                    .get("next")
+                    .and_then(|next| next.parse::<usize>().ok());
+                let room = RequestedRoom { id: room_id, next };
```

Comment:
Is there a way we could layer these two together, thus exposing "room_by_name" and "next_room" functionality individually? And possibly include them in `matchbox_signaling`. Perhaps we could borrow (or even make use of) `tower`? Don't have a clear picture of how this would look yet, but would be nice API wise

Reply:
Not sure what you mean.

`connection` is an object with `.path` and `.query_params` extracted through `tower`. We use the path as the `RoomId` and get `?next=..` for the next room. 

I don't really love the idea of providing `connection.next` or `connection.room` (if that's what you mean) since it's a bit opinionated.

Reply:
Ah, this was an early formation of the thought regarding nested handlers. Can close this thread as the main comment and [this comment](https://github.com/johanhelsing/matchbox/pull/225#discussion_r1234494081) present the idea in a far more comprehensible manner